src/service/viewer-impl.js

```diff
@@ -791,27 +797,27 @@ export class Viewer {
   /**
    * Adds a eventType listener for viewer events.
    * @param {string} eventType
-   * @param {function(!JsonObject)} handler
+   * @param {!RequestHandler} handler
    * @return {!UnlistenDef}
    */
   onMessage(eventType, handler) {
-    let observable = this.messageObservables_[eventType];
-    if (!observable) {
-      observable = new Observable();
-      this.messageObservables_[eventType] = observable;
-    }
-    return observable.add(handler);
+    this.messageHandlers_[eventType] = handler;
```

Comment:
Does this mean that a given eventType can only have one handler? Not sure that's desirable or safe -- let's retain the current behavior if possible.

How about adding an optional parameter to `Observable#fire` such that it returns `Promise.all(<invoked_handlers>)` if provided?

Reply:
I don't think that would work because then the promise would return an array, and that's not what we want to send in the response. I updated this code now to keep the `onMessage` method as is, and add a separate `onMessageRespond` method that makes the provided function the sole handler that responds to the message. So any given message type can have one responder, but multiple other observers.